cmd/gb-vendor/delete.go

```diff
@@ -9,41 +10,69 @@ import (
 	"github.com/constabulary/gb/cmd/gb-vendor/vendor"
 )
 
+var (
+	// gb vendor delete flags
+
+	// delete all dependencies
+	deleteAll bool
+)
+
 func init() {
 	registerCommand("delete", DeleteCmd)
 }
 
+func addDeleteFlags(fs *flag.FlagSet) {
+	fs.BoolVar(&deleteAll, "all", false, "delete all dependencies")
+}
+
 var DeleteCmd = &cmd.Command{
 	ShortDesc: "deletes a local dependency",
 	Run: func(ctx *gb.Context, args []string) error {
-		if len(args) != 1 {
-			return fmt.Errorf("delete: import path missing")
+		if len(args) != 1 && !deleteAll {
+			return fmt.Errorf("delete: import path or --all flag is missing")
+		} else if len(args) == 1 && deleteAll {
+			return fmt.Errorf("delete: you cannot specify path and --all flag at once")
 		}
-		path := args[0]
 
 		m, err := vendor.ReadManifest(manifestFile(ctx))
 		if err != nil {
 			return fmt.Errorf("could not load manifest: %T %v", err, err)
 		}
 
-		d, err := m.GetDependencyForImportpath(path)
-		if err != nil {
-			return fmt.Errorf("could not get dependency: %T %v", err, err)
+		var dependencies []vendor.Dependency
+		if deleteAll {
+			dependencies = make([]vendor.Dependency, len(m.Dependencies))
+			copy(dependencies, m.Dependencies)
+		} else {
+			p := args[0]
+			dependency, err := m.GetDependencyForImportpath(p)
+			if err != nil {
+				return fmt.Errorf("could not get dependency: %T %v", err, err)
+			}
+			dependencies = append(dependencies, dependency)
 		}
 
-		err = m.RemoveDependency(d)
-		if err != nil {
-			return fmt.Errorf("dependency could not be deleted: %T %v", err, err)
-		}
+		for _, d := range dependencies {
+			path := d.Importpath
 
-		localClone := vendor.GitClone{
-			Path: filepath.Join(ctx.Projectdir(), "vendor", "src", path),
-		}
-		err = localClone.Destroy()
-		if err != nil {
-			return fmt.Errorf("dependency could not be deleted: %T %v", err, err)
-		}
+			if err != nil {
+				return fmt.Errorf("could not get dependency: %T %v", err, err)
+			}
+
+			err = m.RemoveDependency(d)
+			if err != nil {
+				return fmt.Errorf("dependency could not be deleted: %T %v", err, err)
```

Comment:
opps, that looks like that's left over debugging. Could you fix it for me ?

On Tue, May 26, 2015 at 9:50 PM, Dávid Kaya <notifications@github.com>
wrote:

> In cmd/gb-vendor/delete.go
> <https://github.com/constabulary/gb/pull/134#discussion_r31025416>:
>
> >
> > -		localClone := vendor.GitClone{
> > -			Path: filepath.Join(ctx.Projectdir(), "vendor", "src", path),
> > -		}
> > -		err = localClone.Destroy()
> > -		if err != nil {
> > -			return fmt.Errorf("dependency could not be deleted: %T %v", err, err)
> > -		}
> > +			if err != nil {
> > +				return fmt.Errorf("could not get dependency: %T %v", err, err)
> > +			}
> > +
> > +			err = m.RemoveDependency(d)
> > +			if err != nil {
> > +				return fmt.Errorf("dependency could not be deleted: %T %v", err, err)
>
> @davecheney <https://github.com/davecheney>
> It is %T %v, I used the same format as you used in the fetch command so
> the error messages won't differ.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/constabulary/gb/pull/134/files#r31025416>.
>